### ISSUE TITLE
MeasureFunc given incorrect AvailableSpace

### ISSUE THREAD

**Issue by ickshonpe** [DioxusLabs/taffy#406]
## `taffy` version

0.3.9

## What you did

```html
<!DOCTYPE html>
<html lang="en">
<head>
  <script src="../scripts/gentest/test_helper.js"></script>
  <link rel="stylesheet" type="text/css" href="../scripts/gentest/test_base_style.css">
  <title>
    Test description
  </title>
<head/>
<body>

<div id="test-root" style="display: flex; width: 100px;">
    <div style="min-width: 200px;">
      01234567&ZeroWidthSpace;01234567    
    </div>
</div>
```

</body>
</html>

## What went wrong

```
Computed tree:
TREE
└──  FLEX [x: 0    y: 0    width: 100  height: 20  ] (2v1)
    └──  LEAF [x: 0    y: 0    width: 200  height: 20  ] (1v1)
```

The width of the inner node is calculated correctly, but the height should be 10 (the height of a single unwrapped line).
It seems like the MeasureFunc is given the wrong AvailableSpace width value (100 rather than 200).

**Comment by nicoburns** [comment]
I think you're running into a limitation of the test generator rather than an actual bug here. The only valid text characters within a gentest node are `H` and `\u{200b}` (zero-width space), and *this includes whitespace*. Your test above is generating the following code:

```rust
const TEXT: &str = "\n      01234567\u{200b}01234567    \n    ";
```

Which is going to result 3 "words" of:
- "\n      01234567" (15 width)
- "      01234567\n.   " (17 width)

The test generator further assumes that every character except `\u{200b}` is `10px` wide (which is true for `H` with the font size and family we have configured). Thus you're getting 150px + 170px > 200px and thus it wraps.

If I change the test to:
```html
<div id="test-root" style="display: flex; width: 100px;">
    <div style="min-width: 200px;">HHHHHHHH&ZeroWidthSpace;HHHHHHHH</div>
</div>
```

which I think captures your original intent of generating two 80px wide words, then it passes.

**Comment by ickshonpe** [comment]
oh no :sweat:. I hacked the minimal html test above down from a more complex example in Bevy and since it showed the same behaviour I assumed it was the same bug and didn't check the final generated test output. It should have been:

```
<div id="test-root" style="display: flex; flex-direction: column; width: 100px;">
    <div style="min-width: 200px;">HHHHHHHH&ZeroWidthSpace;HHHHHHHH</div>
</div>

```
Computed tree:
TREE
└──  FLEX [x: 0    y: 0    width: 100  height: 20  ] (2v1)
    └──  LEAF [x: 0    y: 0    width: 200  height: 20  ] (1v1)
```

**Comment by nicoburns** [comment]
Ah yes, that one I can reproduce. I'll take a look later.

**Comment by nicoburns** [comment]
Fixed in Taffy `0.3.10`. I suspect there are still related issues that this fix doesn't cover, but I don't have time to look into that right now, so I figured it was best to just release this fix as-is.
